KicadModTree/nodes/specialized/PadArray.py

```diff
@@ -44,11 +46,11 @@ class PadArray(Node):
           y offset between rendered pads
         * *initial* (``int``) --
           name of the first pad
-        * *increment* (``int``) --
+        * *increment* (``int, function(previous_number)``) --
```

Comment:
hm, instead of ```function(previous_number)``` an iterator would make more sense

Reply:
or you should give an x/y index and the function returns the id. Would be more general I think and easier to understand.

Reply:
Pad arrays do not create a grid of pads. They only create one line. (Either in x or y direction or diagonal if you give both x and y spacing at the same time.) Meaning using x and y index does not really make sense.

Making it an iterator might be an option. I will look into that. (I however fear that iterators might be a bit above the skill level of the expected users of this library. Maybe i can support it in addition to the simpler function route.)